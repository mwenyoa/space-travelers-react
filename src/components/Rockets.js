import React from 'react';

const Rockets = () => (
  <>
    <section className="rocket-details d-flex">
      <div className="rocket-img-sec"><img src="" alt="Rocket-img" /></div>
      <div className="reserve-rocket-info d-flex flex-d-column">
        <h2>Rocket Name</h2>
        <div className="rocket-desc"><p>Hello World from spacex</p></div>
        <div className="reserve-btn">
          <button type="button">Reserve Rocket</button>
        </div>
      </div>
    </section>
  </>
);
export default Rockets;
