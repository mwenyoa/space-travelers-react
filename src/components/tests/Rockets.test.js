import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../Rockets';
import store from '../../Redux/ConfigureStore';

describe('Rocket Element Tests', () => {
  test('Check if Rockets Component Renders orperly', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(screen.queryByText(/Status/)).toBeNull();
    expect(screen.queryByText(/Description/)).toBeNull();
  });
  test('renders porperly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
