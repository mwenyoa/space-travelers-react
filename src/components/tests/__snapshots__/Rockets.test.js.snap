// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Element Tests renders porperly 1`] = `null`;
